compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -739,11 +739,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                             // So this is a pointer then, and casting to an int
                             // failed. Can only happen during CTFE. If the magic
                             // value is 0 and the scalar is not null, we know
-                            // this is OK. Anything else we conservatively
-                            // reject.
-                            let ptr_valid =
+                            // the pointer cannot be the magic value. Anything
+                            // else we conservatively reject.
+                            let ptr_definitely_not_in_niche_variants =
```

Comment:
Nono this is the flag one, not the niche variants one. ;)

Reply:
This should be ptr_definitely_not_magic_val.

The *other* one should be ptr_definitely_not_in_niche_variants.

Reply:
oh oops!